pages/docs/recipes/adding-a-left-panel.mdx

```diff
@@ -0,0 +1,284 @@
+  - `basePath`: The base path of the app (`/openmrs/spa` by default). This is used to ensure that the left panel links are relative to the app's base path.
+
+- We're calling the `unsetLeftNav` function in a `useEffect` cleanup function. This function takes the name of the slot that the left panel was rendered in. This ensures that the left panel gets unregistered when the component unmounts.
+
+- We're rendering the `LeftNavMenu` component as the first child of the `BrowserRouter` component. This ensures that the left panel gets rendersd in all the routes of the app.
+
+- We're setting up three routes for the Bed Management app:
+
+  - A `/summary` route that renders the `Home` component.
+  - A `/ward/:wardUuid` route that renders the `WardWithBeds` component. The `:ward` portion of the route is a URL parameter that gets passed to the `WardWithBeds` component as a prop. This component is used to render the detail page for a specific ward.
+  - An `/ward-allocation` route that renders the `BedAdministrationTable` component.
+
+### Step 2: Wiring up the root page
+
+Next, we'll create a named export for the `Root` component inside the `index.ts` file. This is the component that we'll use to wire up the `Root` component to the `bed-management` route so that it gets rendered when you navigate to that route.
+
+Add the following to your `index.ts` file:
+
+```ts copy
+// src/index.ts
+export const root = getAsyncLifecycle(
+  () => import("./root.component"),
+  options
+);
+```
+
+Next, modify your `routes.json` file to include the following page definition:
+
+```json {7-12} copy
+{
+  "$schema": "https://json.openmrs.org/routes.schema.json",
+  "backendDependencies": {
+    "fhir2": "^1.2.0",
+    "webservices.rest": "^2.24.0"
+  },
+  "pages": [
+    {
+      "component": "root",
+      "route": "bed-management"
+    }
+  ]
+}
+```
+
+This page definition tells O3 to render the `Root` component when you navigate to the `bed-management` route.
+
+### Step 3: Add links to the left panel
+
+Next, we'll want to add the two links that we mentioned earlier to the left panel:
+
+- A `Summary` link for the landing screen.
+- A `Ward allocation` link for the ward allocation screen.
+
+To do this, we'll need to create two extensions that render the links. We'll then add those extensions to the `extensions` array of our `routes.json` file. The named exports of those extensions will be used as the `component` property of the extensions, and look like the following:
+
+```ts copy
+// src/index.ts
+export const summaryLeftPanelLink = getSyncLifecycle(
+  createLeftPanelLink({
+    name: "summary",
+    title: t("summary", "Summary"),
+  }),
+  options
+);
+
+export const wardAllocationLeftPanelLink = getSyncLifecycle(
+  createLeftPanelLink({
+    name: "ward-allocation",
+    title: t("wardAllocation", "Ward Allocation"),
+  }),
+  options
+);
+```
```

Comment:
You will have to pass the `moduleName` in the useTranslation hook in the code.

Reply:
@vasharma05 Note that the code here _isn't_ using the `createDashboardLink()` function from the patient chart, so the requirements here are a bit different. Also, please note that you _only_ need comments with the `t` function if:

1. The translation is needed at runtime but otherwise doesn't appear in code
2. The key to be translated cannot be inferred from the code

Neither of those two apply here.